teps/0076-array-result-types.md

```diff
@@ -465,6 +484,19 @@ tasks:
       - name: environments
         value: '$(tasks.get-environments.results.environments[*])'
 ```
+A third way for variable replacements can be:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+...
+spec:
+tasks:
+  - name: update-all-environments
+    params:
+      - name: environments
+        value:
+          - '$(tasks.get-environments.results.environments[*])'
```

Comment:
I can see another benefit of supporting this in array is that we can append several results. But for Object it may not be so helpful? @chuangw6 thinks we shouldn't add support for this

Reply:
For array, this variable replacement makes sense to me. 

For object, it's different. I'll discuss with you offline.

Reply:
Yes. I should've be more clear about this, but since 0076 is all around arrays, so I will only focus on examples for arrays. I will open another PR for 0075 if necessary, once we finalize our discussions in https://github.com/tektoncd/pipeline/issues/4879

Reply:
FWIW, I'd interpret this syntax as "an array element that holds an array", which is different than the previous 2 examples. Nested arrays are a non-goal for this PR, so perhaps we should omit this?